modules/issue_tracker/php/edit.class.inc

```diff
@@ -0,0 +1,859 @@
+            );
+            if (!$validCenter) {
+                $validCenter = $db->pselectOne(
+                    "
+                    SELECT
+                        EXISTS (
+                            SELECT
+                                *
+                            FROM
+                                session s
+                            JOIN
+                                candidate c
+                            ON
+                                c.CandID = s.CandID
+                            WHERE
+                                s.CenterID = :center_id AND
+                                c.PSCID = :psc_id
+                        )
+                ",
+                    [
+                        "center_id" => $result['centerID'],
+                        "psc_id"    => $result['PSCID'],
+                    ]
+                );
+            }
+            if (!$validCenter) {
+                error_log('PSCID and Center ID do not match a valid session!');
+                return new \LORIS\Http\Response\JSON\InternalServerError(
+                    'PSCID and Center ID do not match a valid session!'
+                );
+            }
+        }
+        // If both are set, return SessionID and CandID
+        if (isset($result['PSCID']) && isset($result['visit'])) {
+            $session = $db->pSelect(
+                "SELECT s.ID as sessionID, c.candID as candID FROM candidate c
+                INNER JOIN session s on (c.CandID = s.CandID)
+                WHERE c.PSCID=:PSCID and s.Visit_label=:visitLabel",
+                [
+                    'PSCID'      => $result['PSCID'],
+                    'visitLabel' => $result['visit'],
+                ]
+            );
+
+            if (isset($session[0]['sessionID'])) {
+                $result['sessionID'] = $session[0]['sessionID'];
+                $result['candID']    = $session[0]['candID'];
+            } else {
+                error_log(
+                    'PSCID and Visit Label do not match a valid candidate session!'
+                );
+                return new \LORIS\Http\Response\JSON\InternalServerError(
+                    'PSCID and Visit Label do not match a valid candidate session!'
+                );
+            }
+            return $result;
+        }
+
+        // If only PSCID is set, return CandID
+        if (isset($result['PSCID'])) {
+            $query  = "SELECT CandID FROM candidate WHERE PSCID=:PSCID";
+            $params = ['PSCID' => $result['PSCID']];
+
+            if (!$user->hasPermission('access_all_profiles')) {
+                $params['CenterID'] = implode(',', $user->getCenterIDs());
+                $query .= " AND FIND_IN_SET(RegistrationCenterID,:CenterID)";
+            }
+
+            $candidate = $db->pSelectOne($query, $params);
+            if ($candidate) {
+                $result['candID'] = $candidate;
+            } else {
+                error_log('PSCID does not match a valid candidate!');
```

Comment:
I don't see why this is going to the error log, if it needs to it should be using $this->logger